test/integration/simulacron.js

```diff
@@ -111,200 +163,352 @@ var simulacronHelper = {
+        }
       }
     }
-  }
-
-  for(var dcIndex = 0; dcIndex < self.dcs.length; dcIndex++) {
-    var nodeFound = findInDc(self.dcs[dcIndex]);
-    if (nodeFound) {
-      return nodeFound;
+    // node not found, raise error.
+    throw new Error("No node found for " + args[0]);
+  } else {
+    var dcId = 0;
+    var nodeId;
+    // if only one argument provided, assume nodeId in 0th data center.
+    if (args.length === 1) {
+      nodeId = args[0];
+    } else {
+      dcId = args[0];
+      nodeId = args[1];
     }
+    dc = this.dcs[dcId];
+    return dc.node(nodeId);
   }
 };
 
+/**
+ * Finds a data center in the cluster by its id.
+ * 
+ * @param {Number} id Idenfitifer of the dc.
+ * @returns {SimulacronDataCenter} The data center, if found.
+ */
+SimulacronCluster.prototype.dc = function(id) {
+  return this.dcs[id];
+};
+
+/**
+ * @param {Number} dataCenterId The data center to return node addresses from, if not provided assumes 0.
+ * @returns {Array} Listing of addresses of the nodes in the input dc.
+ */
 SimulacronCluster.prototype.getContactPoints = function(dataCenterId) {
   var dcId = dataCenterId = typeof dataCenterId === 'undefined' ? 0 : dataCenterId;
   return this.dcs[dcId].nodes.map(function (node) {
     return node.address;
   });
 };
 
-SimulacronCluster.prototype.queryNodeLog = function(nodeAddress, callback) {
+/**
+ * Represents a data center with its associated node configurations. 
+ * @param {SimulacronCluster} cluster Parent cluster.
+ * @param {Object} dc json data from provision response.
+ */
+function SimulacronDataCenter(cluster, dc) {
+  SimulacronTopic.call(this);
+  this.cluster = cluster;
+  this.data = dc;
+  this.localId = dc.id;
+  this.id = cluster.id + '/' + dc.id;
   var self = this;
-  var node = self.findNode(nodeAddress);
-  if (node === null) {
-    throw new Error('invalid node address: ' + nodeAddress);
-  }
+  this.nodes = dc.nodes.map(function(node) {
+    return new SimulacronNode(self, node);
+  });
+}
 
-  var queryLog = '/log/%d/%d/%d';
-  var options = {
-    host: self.baseAddress,
-    path: encodeURI(util.format(queryLog, self.id, node.dataCenterId, node.nodeId)),
-    port: self.port,
-    method: 'GET'
-  };
-  makeRequest(options, function(data) {
-    callback(data.data_centers[0].nodes[0].queries);
-  }).end();
-};
+util.inherits(SimulacronDataCenter, SimulacronTopic);
 
-SimulacronCluster.prototype.stopNode = function(nodeAddress, callback) {
-  var self = this;
-  var node = self.findNode(nodeAddress);
-  if (node === null) {
-    callback(new Error('invalid node address: ' + nodeAddress));
-  }
+/**
+ * Finds a node in the given data center by its id or address.
+ * 
+ * @param {Number|String} id Identifier of node.  If Number, assumes the id of the node in the data center.
+ * If String, assumes an 'ip:port' designation and looks up node by address.
+ * @returns {SimulacronNode} The node, if found.
+ */
+SimulacronDataCenter.prototype.node = function() {
```

Comment:
Use the actual parameter field instead of `arguments` array whenever possible (for IDE support, param doc definition, code validation like eslint, ...)